frame/contracts/src/wasm/mod.rs

```diff
@@ -895,6 +920,84 @@ mod tests {
 		assert_eq!(u32::from_le_bytes(result.data.0.try_into().unwrap()), 0,);
 	}
 
+	#[test]
+	#[cfg(feature = "unstable-interface")]
+	fn contains_storage_works() {
+		const CODE: &str = r#"
+(module
+	(import "seal0" "seal_return" (func $seal_return (param i32 i32 i32)))
+	(import "seal0" "seal_input" (func $seal_input (param i32 i32)))
+	(import "__unstable__" "seal_contains_storage" (func $seal_contains_storage (param i32 i32) (result i32)))
+	(import "env" "memory" (memory 1 1))
+
+
+	;; size of input buffer
+	;; [0, 4) size of input buffer (128+32 = 160 bytes = 0xA0)
+	(data (i32.const 0) "\A0")
+
+	;; [4, 164) input buffer
+
+	(func (export "call")
+		;; Receive key
+		(call $seal_input
+			(i32.const 4)	;; Where we take input and store it
+			(i32.const 0)	;; Where we take and store the length of the data
+		)
+		;; Call seal_clear_storage and save what it returns at 0
+		(i32.store (i32.const 0)
+			(call $seal_contains_storage
+				(i32.const 8)			;; key_ptr
+				(i32.load (i32.const 4))	;; key_len
+			)
+		)
+		(call $seal_return
+			(i32.const 0)	;; flags
+			(i32.const 0)	;; returned value
+			(i32.const 4)	;; length of returned value
+		)
+	)
+
+	(func (export "deploy"))
+)
+"#;
+
+		let mut ext = MockExt::default();
+		ext.set_storage_transparent(
+			VarSizedKey::<Test>::try_from([1u8; 64].to_vec()).unwrap(),
+			Some(vec![42u8]),
+			false,
+		)
+		.unwrap();
+		ext.set_storage_transparent(
+			VarSizedKey::<Test>::try_from([2u8; 19].to_vec()).unwrap(),
+			Some(vec![]),
+			false,
+		)
+		.unwrap();
+
+		//value does not exist (wrong key length)
+		let input = (63, [1u8; 64]).encode();
```

Comment:
But you could just use a shorter input in this case?:
```rust
let input = [1u8; 63].encode();
```

Reply:
this would work only for the case when the passed `key_len` is wrong, as
```rust
let input = [1u8; 63].encode(); // key_len = "\49\02" (585u32), key = [1u8;59]: wrong key_len (585 != 59)
```
while other cases (correct `key_len`) would still need it like this
```rust
let input = (64, [1u8; 63]).encode();
```
so I decided to keep it uniformly along all cases, for better readability.

Reply:
Why? You can use the length written by `seal_input` in both cases. In one case you write a too short of an input and in the other case you don't. You don't need to separately signal the size of the buffer. `seal_input` communicates that.

Reply:
But if you pass `[1u8; 63]`  instead of `(63, [1u8; 64]).encode()` wouldn't you be testing the same thing? `seal_input` would communicate `63` as `key_len`.

Reply:
If I pass `[1u8; 63]` I would test versus `[1u8; 63]` `key` and `63` `key_len`. 
I pass `(63, [1u8; 64])` to test versus `[1u8; 64]` `key` and `63` `key_len`.

Reply:
Okay, I see why this tests more things. Like some bug where the key length is not properly honored.